sim/hunter/rapid_fire.go

```diff
@@ -24,13 +24,12 @@ func (hunter *Hunter) registerRapidFireCD() {
 		OnGain: func(aura *core.Aura, sim *core.Simulation) {
 			aura.Unit.MultiplyRangedSpeed(sim, hasteMultiplier)
 
-			if manaMetrics != nil {
-				manaPerTick := 0.02 * float64(hunter.Talents.RapidRecuperation) * hunter.MaxMana()
+			if focusMetrics != nil {
 				core.StartPeriodicAction(sim, core.PeriodicActionOptions{
```

Comment:
I know you havent changed this but its a bug fix i did for other periodic actions like this one.
define a PA var above the aura and in the OnGain assing the periodic action you start to this var and on the expire do `pa.Cancel(sim)` to make sure if for some reason the aura is removed before its original expiration time it properly stops the regen as well (I had to fix a couple of those on the SoD fork)

Reply:
Check my implementation and see how it looks